src/app/services/mail/mail.service.ts

```diff
@@ -157,13 +167,36 @@ export class MailService {
       })
 
       try {
-        const info = await tracer.trace('nodemailer/sendMail', () =>
-          this.#transporter.sendMail(mail),
-        )
+        const getThreshold = () => {
+          const now = Date.now()
+          const elapsed = Math.max(
+            0,
+            Math.min(now - START_TS, WARM_UP_DURATION),
+          )
+          const linear = elapsed / WARM_UP_DURATION
+          return linear * linear
+        }
```

Comment:
/nit this function is not dependent on the call context (it only relies on `START_TS` and `WARM_UP_DURATION`, which are module constants), so the function itself can be declared just once at module level.

And then maybe just for extra clarity the function can be renamed `getWarmUpThreshold()` 😅

Reply:
[c77a6cd](https://github.com/opengovsg/FormSG/pull/5369/commits/c77a6cd2f529c179152ed3f9be395ab139d57cdf)

Reply:
Hey? I'm not seeing that change in the commit? did you forget to add the file? 😅

Reply:
nvm, I see them in the second "TODO" commit, oops!